### PR TITLE
Main bc owl mainnet 3

### PR DESCRIPTION
**Pull request by fluidvanadium**
> (optional) to update the hot wallet in order to minimize execution time:
> 
> cargo run -- --data-dir zingolib/src/wallet/disk/testing/examples/mainnet/hhcclaltpcckcsslpcnetblr/latest/  
> 
> 
> to run a basic hot test:
> 
> cargo nextest run --run-ignored=all mainnet_latest_send_to_self_orchard_hot

test only PR

- mainnet tests
- separated assert_fee_and_record into helpers
- numerous micro refactors

### CODE REVIEW

File: zingolib/src/lightclient/send.rs [hunk 1]
```diff
@@ -405,20 +414,166 @@ pub mod send_with_proposal {
...
-            ))
-            .await;
+                ));
+
+            let client = sync_example_wallet(case).await;
 
             with_assertions::propose_shield_bump_sync(&mut LiveChain::setup().await, &client, true)
                 .await;
         }
+
```
Comment (zancas) [inline zingolib/src/lightclient/send.rs:424]: So should we ignore this?   It shouldn't be part of the automated test suite, should it?

File: zingolib/src/lightclient/send.rs [hunk 2]
```diff
@@ -405,20 +414,166 @@ pub mod send_with_proposal {
...
+        /// this is a live sync test. its execution time scales linearly since last updated
+        /// this is a live send test. whether it can work depends on the state of live wallet on the blockchain
+        /// note: live send waits 2 minutes for confirmation. expect 3min+ runtime
+        async fn mainnet_latest_send_to_self_orchard_hot() {
+            // install default crypto provider (ring)
+            if let Err(e) = rustls::crypto::ring::default_provider().install_default() {
+                log::error!("Error installing crypto provider: {:?}", e)
+            };
+
+            let case = ExampleWalletNetwork::Mainnet(crate::wallet::disk::testing::examples::ExampleMainnetWalletSeed::HHCCLALTPCCKCSSLPCNETBLR(crate::wallet::disk::testing::examples::ExampleHHCCLALTPCCKCSSLPCNETBLRVersion::Latest));
```
Comment (zancas) [inline zingolib/src/lightclient/send.rs]: This is unreadable..   right?

File: zingolib/src/lightclient/send.rs [hunk 3]
```diff
@@ -405,20 +414,166 @@ pub mod send_with_proposal {
...
+        /// this is a live send test. whether it can work depends on the state of live wallet on the blockchain
+        /// note: live send waits 2 minutes for confirmation. expect 3min runtime
+        #[ignore = "dont automatically run hot tests! this test spends actual zec!"]
+        async fn mainnet_hhcclaltpcckcsslpcnetblr_send_to_self_sapling_hot() {
+            // install default crypto provider (ring)
+            if let Err(e) = rustls::crypto::ring::default_provider().install_default() {
+                log::error!("Error installing crypto provider: {:?}", e)
+            };
+
+            let case = ExampleWalletNetwork::Mainnet(crate::wallet::disk::testing::examples::ExampleMainnetWalletSeed::HHCCLALTPCCKCSSLPCNETBLR(crate::wallet::disk::testing::examples::ExampleHHCCLALTPCCKCSSLPCNETBLRVersion::Latest));
```
Comment (zancas) [inline zingolib/src/lightclient/send.rs]: This is not readable.

File: zingolib/src/testutils/chain_generics/with_assertions.rs [hunk 4]
```diff
@@ -1,10 +1,10 @@
 //! lightclient functions with added assertions. used for tests.
 
-use crate::lightclient::LightClient;
+use crate::{lightclient::LightClient, testutils::lightclient::lookup_stati};
```
Comment (zancas) [inline zingolib/src/testutils/chain_generics/with_assertions.rs]: What's a "lookup_stati"?

File: zingolib/src/testutils/lightclient.rs [hunk 5]
```diff
@@ -88,8 +89,25 @@ pub mod from_inputs {
...
         crate::wallet::propose::ProposeSendError,
     > {
+        // TOdo fix expect
         let request = transaction_request_from_send_inputs(raw_receivers)
             .expect("should be able to create a transaction request as receivers are valid.");
         proposer.propose_send(request).await
     }
 }
+
+/// gets stati for a vec of txids
```
Comment (zancas) [inline zingolib/src/testutils/lightclient.rs:99]: Please change "stati" into a word or phrase that means something in English.

**Review by zancas — CHANGES_REQUESTED** [review]
Please reformat any lines that unreadably long.
Consider ignoring all mainnet tests by default.

File: zingolib/src/testutils/chain_generics/with_assertions.rs (hunk 4 above)
Comment (fluidvanadium) [reply zingolib/src/testutils/chain_generics/with_assertions.rs]: it is doc-commented

File: zingolib/src/lightclient/send.rs (hunk 1 above)
Comment (fluidvanadium) [reply zingolib/src/lightclient/send.rs:424]: yeahh

**Review by zancas — DISMISSED** [review]
Please change from initial letter on each token of the seed phrase to the initial tokens in the seed..  maybe 2 or 3.

Please use the default import syntax that code actions enforce on files.

**Comment by fluidvanadium** [comment]
> Please change from initial letter on each token of the seed phrase to the initial tokens in the seed.. maybe 2 or 3.
> 
> Please use the default import syntax that code actions enforce on files.

not within the scope of this PR

**Comment by zancas** [comment]
> > Please change from initial letter on each token of the seed phrase to the initial tokens in the seed.. maybe 2 or 3.
> > Please use the default import syntax that code actions enforce on files.
> 
> not within the scope of this PR

Yeah, I am fine with these issues being addressed in another PR.  That could be a cleaner flow.

File: zingolib/src/testutils/lightclient.rs [hunk 6]
```diff
@@ -88,8 +89,25 @@ pub mod from_inputs {
...
+pub async fn lookup_stati(
```
Comment (AloeareV) [inline zingolib/src/testutils/lightclient.rs:100]: statuses would be immediately understood, whereas stati, though I figured out what it meant, is not as easily understandable.
```suggestion
pub async fn lookup_statuses(
```

File: zingolib/src/testutils/assertions.rs [hunk 7]
```diff
@@ -29,27 +49,28 @@ pub async fn assert_record_fee_and_status<NoteId>(
...
-    assert_eq!(proposal.steps().len(), txids.len());
-    let mut total_fee = 0;
-    for (i, step) in proposal.steps().iter().enumerate() {
-        let record = records.get(&txids[i]).unwrap_or_else(|| {
-            panic!(
-                "sender must recognize txid.\nExpected {}\nRecognised: {:?}",
-                txids[i],
-                records.0.values().collect::<Vec<_>>()
-            )
+    let mut step_results = vec![];
```
Comment (AloeareV) [inline zingolib/src/testutils/assertions.rs:52]: Much more flexible. Nice!

**Review by AloeareV — DISMISSED** [review]

File: zingolib/src/testutils/lightclient.rs (hunk 6 above)
Comment (zancas) [reply zingolib/src/testutils/lightclient.rs:100]: I had the exact same problem.

**Pull request merged**